providers/mongodb.js

```diff
@@ -85,3 +85,18 @@ module.exports = class extends Provider {
 };
 
 const resolveQuery = query => isObject(query) ? query : { id: query };
+
+const parseUpdateObject = (doc, pref = '', oldObj = {}) => {
```

Comment:
This feels very wrong, should check whether doc is an object, or the output from the engine (which is an array).